config/config.go

```diff
@@ -74,6 +75,31 @@ func WithType(in string) LoaderOption {
 	}
 }
 
+// WithCobraBindFlags binds cli flags to all `cmdx`
+// tags in the struct. Use tag `cmdx` to bind struct
+// field to cli flag.
+// e.g.
+//
+//	type Config struct {
+//		Host string `yaml:"host" cmdx:"host"`
+//	}
+func WithCobraBindFlags(cmd *cobra.Command, cfg interface{}) LoaderOption {
```

Comment:
Since we are only using `cmd.Flags()`, Can we instead pass that as `pflag.FlagSet`?

Reply:
I think that is doable, let me try that @rohilsurana